### ISSUE TITLE
 EIP-712 domain.chainId is interpreted as hex and cause eth_signTypedData_v4 chainId mismatch error

### ISSUE THREAD

**Issue by PierreJeanjacquot** [MetaMask/metamask-extension#11830]
**Describe the bug**
 EIP-712 domain.chainId is interpreted as hex and cause eth_signTypedData_v4 chainId mismatch error.

**Steps to reproduce (REQUIRED)**
MetaMask 10.0.1
1. Open live demo: https://codesandbox.io/s/metamask-10-0-1-eip-712-issue-nnxvc?file=/src/index.js
2. Switch metamask network to Kovan an reload
3. see error in console

**Expected behavior**
considering the following typed data,`"chainId": "42"` should be parsed as decimal (as it use to be) and eth_signTypedData_v4 should work 
```json
{
  "domain": {
    "chainId": "42"
  },
  "types": {
    "Foo": [
      {
        "name": "bar",
        "type": "uint"
      }
    ]
  },
  "primaryType": "Foo",
  "message": {
    "bar": "10"
  }
}
```

**Browser details (please complete the following information):**
 - OS: Ubuntu 20.04
 - Browser: Firefox Browser 91.0 (64-bit)
 - MetaMask Version: 10.0.1

**Comment by PierreJeanjacquot** [comment]
should be fixed by https://github.com/MetaMask/metamask-extension/pull/11900

**Comment by PierreJeanjacquot** [comment]
fixed in MetaMask 10.0.3
